circuit/src/queryBuilderBase/queryBuilderBase.ts

```diff
@@ -0,0 +1,465 @@
+          break;
+        case DataSubqueryType.Transaction:
+          validQuery =
+            validQuery && (await validateTxSubquery(provider, subquery as TxSubquery, configLimitManager));
+          break;
+        case DataSubqueryType.Receipt:
+          validQuery =
+            validQuery &&
+            (await validateReceiptSubquery(provider, subquery as ReceiptSubquery, configLimitManager));
+          break;
+        case DataSubqueryType.SolidityNestedMapping:
+          validQuery =
+            validQuery &&
+            (await validateSolidityNestedMappingSubquery(provider, subquery as SolidityNestedMappingSubquery));
+          break;
+        case DataSubqueryType.BeaconValidator:
+          validQuery =
+            validQuery && (await validateBeaconSubquery(provider, subquery as BeaconValidatorSubquery));
+          break;
+        default:
+          throw new Error(`Invalid subquery type: ${type}`);
+      }
+    }
+    return validQuery;
+  }
+
+  protected async validateComputeQuery(): Promise<boolean> {
+    if (this.computeQuery === undefined) {
+      return true;
+    }
+    let valid = true;
+
+    // Check resultLen
+    if (
+      this.computeQuery.resultLen !== undefined &&
+      this.computeQuery.resultLen > AxiomV2CircuitConstant.UserMaxOutputs
+    ) {
+      console.warn(`Callback resultLen is greater than maxOutputs (${AxiomV2CircuitConstant.UserMaxOutputs})`);
+      valid = false;
+    }
+
+    // Check that vkey and computeProof are not zero if k is nonzero
+    if (this.computeQuery.k !== 0) {
+      if (this.computeQuery.vkey.length === 0) {
+        console.warn("Compute query vkey is empty");
+        valid = false;
+      }
+      if (this.computeQuery.computeProof.length === 0) {
+        console.warn("Compute query computeProof is empty");
+        valid = false;
+      }
+    }
+
+    return valid;
+  }
+
+  protected resetSubqueryCount() {
+    this.dataSubqueryCount = deepCopyObject(ConstantsV2.EmptyDataSubqueryCount);
+  }
+
+  protected updateSubqueryCount(type: DataSubqueryType, skipValidate?: boolean) {
+    this.dataSubqueryCount.total++;
+    if (skipValidate) return;
+    if (this.dataSubqueryCount.total > ConstantsV2.UserMaxTotalSubqueries) {
+      throw new Error(`Cannot add more than ${ConstantsV2.UserMaxTotalSubqueries} subqueries`);
+    }
+    switch (type) {
+      case DataSubqueryType.Header:
+        this.dataSubqueryCount.header++;
+        if (this.dataSubqueryCount.header > ConstantsV2.MaxSameSubqueryType) {
+          throw new Error(`Cannot add more than ${ConstantsV2.MaxSameSubqueryType} Header subqueries`);
+        }
+        break;
+      case DataSubqueryType.Account:
+        this.dataSubqueryCount.account++;
+        if (this.dataSubqueryCount.account > ConstantsV2.MaxSameSubqueryType) {
+          throw new Error(
+            `Cannot add more than ${ConstantsV2.MaxSameSubqueryType} Account + Storage + Nested Mapping subqueries`,
```

Comment:
is this message outdated?

Reply:
do we want to keep the same subquery type check or just check total number since the values are the same now?